python/ray/experimental/multiprocessing/pool.py

```diff
@@ -0,0 +1,493 @@
+        self._initializer = initializer
+        self._initargs = initargs
+        self._maxtasksperchild = maxtasksperchild or -1
+        self._actor_deletion_ids = []
+
+        processes = self._init_ray(processes, ray_address)
+        self._start_actor_pool(processes)
+
+    def _init_ray(self, processes=None, ray_address=None):
+        # Initialize ray. If ray is already initialized, we do nothing.
+        # Else, the priority is:
+        # ray_address argument > RAY_ADDRESS > start new local cluster.
+        if not ray.is_initialized():
+            if ray_address is None and RAY_ADDRESS_ENV in os.environ:
+                ray_address = os.environ[RAY_ADDRESS_ENV]
+
+            # Cluster mode.
+            if ray_address is not None:
+                print("Connecting to ray cluster at address='{}'".format(
+                    ray_address))
+                ray.init(address=ray_address)
+            # Local mode.
+            else:
+                print("Starting local ray cluster")
+                ray.init(num_cpus=processes)
+
+        ray_cpus = int(ray.state.cluster_resources()["CPU"])
+        if processes is None:
+            processes = ray_cpus
+        if processes <= 0:
+            raise ValueError("Processes in the pool must be >0.")
+        if ray_cpus < processes:
+            raise ValueError("Tried to start a pool with {} processes on an "
+                             "existing ray cluster, but there are only {} "
+                             "CPUs in the ray cluster.".format(
+                                 processes, ray_cpus))
+
+        return processes
+
+    def _start_actor_pool(self, processes):
+        self._actor_pool = [self._new_actor_entry() for _ in range(processes)]
+        ray.get([actor.ping.remote() for actor, _ in self._actor_pool])
+
+    def _wait_for_stopping_actors(self, timeout=None):
+        if len(self._actor_deletion_ids) == 0:
+            return
+        if timeout is not None:
+            timeout = float(timeout)
+
+        _, deleting = ray.wait(
+            self._actor_deletion_ids,
+            num_returns=len(self._actor_deletion_ids),
+            timeout=timeout)
+        self._actor_deletion_ids = deleting
+
+    def _stop_actor(self, actor):
+        # Check and clean up any outstanding IDs corresponding to deletions.
+        self._wait_for_stopping_actors(timeout=0.0)
+        # The deletion task will block until the actor has finished executing
+        # all pending tasks.
+        self._actor_deletion_ids.append(actor.__ray_terminate__.remote())
+
+    def _new_actor_entry(self):
+        # TODO(edoakes): The initializer function can't currently be used to
+        # modify the global namespace (e.g., import packages or set globals)
+        # due to a limitation in cloudpickle.
+        return (PoolActor.remote(self._initializer, self._initargs), 0)
+
+    def _random_actor_index(self):
+        return random.randrange(len(self._actor_pool))
```

Comment:
Can we do something like `return self.rand.randrange(len(self._actor_pool))`, with `self.rand = rand or random.Random()` in the constructor (taking `rand` as a parameter) so the user of the object has better control (and awareness) over the nondeterminism?

Reply:
If users request this it can be added, don't want to expose unnecessary parameters